source/templates/workflows.slim

```diff
@@ -182,4 +190,50 @@
 				button.done[ng-click="workflowsCtrl.rearrangeWorkflowsFinished()"] == data[:strings][:workflows][:rearrange][:done]
 				button.cancel[ng-click="workflowsCtrl.rearrangeWorkflowsPopup.isVisible = false"] == data[:strings][:workflows][:rearrange][:cancel]
 	add-step-sidebar
-	insert-variable-popup
+	popup[model="workflowsCtrl.insertVariableToStepInputConfig.popup"]
+		.popup-body.snowflake#insert-variable-popup-body[ng-if="workflowsCtrl.insertVariableToStepInputConfig.popup.isVisible" ng-controller="InsertVariableController as insertVariableCtrl" ng-init="insertVariableCtrl.init(workflowsCtrl.insertVariableToStepInputConfig.variableSelected, null, workflowsCtrl.insertVariableToStepInputConfig.workflows, workflowsCtrl.insertVariableToStepInputConfig.selectedWorkflow, workflowsCtrl.insertVariableToStepInputConfig.selectedStep)"]
+			header
+				h2 == data[:strings][:workflows][:inputs][:insert_variable][:action]
+				input[type="text" ng-model="insertVariableCtrl.keyFilter" placeholder="#{ data[:strings][:workflows][:inputs][:insert_variable][:key_filter_placeholder] }"]
+			article.vertical-scroll
+				p.progress-indicator[progress-model="insertVariableCtrl.loadInsertableVariablesProgress"]
+				div[ng-if="insertVariableCtrl.loadInsertableVariablesProgress.isIdle"]
+					ul.variable-source[ng-repeat="variablesOfSource in insertVariableCtrl.insertableVariables"]
+						li[ng-repeat="variable in variablesOfSource.variables | variablesfilteredByKey: insertVariableCtrl.keyFilter" ng-if="insertVariableCtrl.shouldShowVariable(variable)"]
+							button[ng-click="insertVariableCtrl.variableSelected(variable)"]
+								strong[ng-bind="variable | prettifiedVariableKey"]
+								em[ng-bind="variablesOfSource.title"]
+	popup[model="workflowsCtrl.insertSecretToStepInputConfig.popup"]
+		.popup-body.snowflake#insert-variable-popup-body[ng-if="workflowsCtrl.insertSecretToStepInputConfig.popup.isVisible" ]
+			header
+				h2 == data[:strings][:workflows][:inputs][:insert_secret][:action]
+			article.vertical-scroll
+				.create-new-secret[ng-controller="CreateAndInsertSecretController as createAndInsertSecretCtrl" ng-init="createAndInsertSecretCtrl.init(workflowsCtrl.insertSecretToStepInputConfig.secretSelected, workflowsCtrl.insertSecretToStepInputConfig.workflows, workflowsCtrl.insertSecretToStepInputConfig.selectedWorkflow, workflowsCtrl.insertSecretToStepInputConfig.selectedStep)"]
```

Comment:
What if user adds a secret to a sensitive input, then removes the secret? The sensitive input becomes invalid.

Reply:
At this point we do not have to handle that case.